### DISCUSSION TITLE
Differentiate LeftShift / RightShift

### DISCUSSION THREAD

**Discussion by rsgjunior** [autokey/autokey#569] [Q&A]
Good evening,

I'd like to know if there's someway to differentiate which SHIFT Key was pressed.

I use a 60% keyboard, so i bind some keys with the RightShift to use as the arrow keys. But, these same keys are used with the LeftShift for other stuff.

The problem is: when i set the hotkey to a script as "shift + anykey" both shift keys activate it.

**Comment by ineuw** [comment]
Autokey doesn't seem to differentiate between the left and right shift key at the moment, but try this link. [link!](https://docs.python.org/3/library/curses.html)

**Reply by josephj11** [reply]
How does the curses module help? I scanned through it and nothing jumped out at me.

**Comment by josephj11** [comment]
AutoKey does not differentiate between the two shift or alt keys.

If you need to do that, then you have  to do it at a lower level by remapping your keyboard with xmodmap or xkb to make one of the two keys do something else. AutoKey isn't particularly fond of remapped keyboards, so you will have to experiment with this to see what works.

If you want to explore this further, I have a bunch of links to relevant articles. Ask about this on our support [list](https://groups.google.com/forum/#!forum/autokey-users).

----

BTW: Whenever you have a question (as opposed to an issue) it's better to post it on our [list](https://groups.google.com/forum/#!forum/autokey-users) or on [Gitter](https://gitter.im/autokey/autokey). The list is more user oriented whereas more technical questions will get a better answer on Gitter.

Both of these venues have a number of active users, so you are likely to get more, better, and faster answers on them than here on GitHub where only our very busy developers and a few users see the posts. We also prefer to have the developers developing rather than taking time to answer questions and manage issues that the user community could handle for them.
